src/recon_buildblock/GeneralisedObjectiveFunction.cxx

```diff
@@ -271,9 +271,8 @@ add_multiplication_with_approximate_sub_Hessian(TargetT& output,
     {
       // TODO used boost:scoped_ptr
       shared_ptr<TargetT>  prior_output_sptr(output.get_empty_copy());
-      if (this->prior_sptr->add_multiplication_with_approximate_Hessian(*prior_output_sptr, output) ==
-	  Succeeded::no)
-	return Succeeded::no;
+      if (this->prior_sptr->accumulate_Hessian_times_input(*prior_output_sptr, output, output) == Succeeded::no)
+	    return Succeeded::no;
```

Comment:
I believe this is the only implementation of the `add_multiplication_with_approximate_Hessian ` for the prior to be replaced with `accumulate_Hessian_times_input`. I assume the usage is correct for the prior.

Reply:
sadly not. The "approximate_Hessian" for the log-likelihood (or in general data-fitting) term is supposed to be roughly independent of the `current_estimate`. We have no idea what the `input` will be. For instance, for OSSPS, `input` is an image with ones
https://github.com/UCL/STIR/blob/b79bb323c1f1abab02f397f04ea8a13995b133c2/src/iterative/OSSPS/OSSPSReconstruction.cxx#L234-L237

Luckily, this function `add_multiplication_with_approximate_sub_Hessian` seems unused in STIR. OSSPS only uses `add_multiplication_with_approximate_sub_Hessian_without_penalty` and the surrogate for the prior
https://github.com/UCL/STIR/blob/29c04d7ac773a8c7ae8911f49aa3b6d6ce447bbc/src/iterative/OSSPS/OSSPSReconstruction.cxx#L377-L378

Maybe we should just remove this function (i.e. have it later with the "true" Hessian one)

Reply:
I have resolved by reverting.